pkgs/development/python-modules/uv-dynamic-versioning/default.nix

```
@@ -0,0 +1,73 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  hatchling,
+
+  # dependencies
+  dunamai,
+  eval-type-backport,
+  pydantic,
+  returns,
+  tomlkit,
+
+  # tests
+  gitpython,
+  pytestCheckHook,
+  writableTmpDirAsHomeHook,
+}:
+
+buildPythonPackage rec {
+  pname = "uv-dynamic-versioning";
+  version = "0.4.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ninoseki";
+    repo = "uv-dynamic-versioning";
+    tag = "v${version}";
+    hash = "sha256-5pc05yazeHyQvmNk0uy6XkLrmX4ZQF6Lj7fDDtVcpu4=";
+    # Tests perform mock operations on the local repo
+    leaveDotGit = true;
```

```suggestion
    # Tests perform mock operations on the local repo
    leaveDotGit = true;
    hash = "sha256-5pc05yazeHyQvmNk0uy6XkLrmX4ZQF6Lj7fDDtVcpu4=";
```
leaveDotGit influeneces the hash